src/other/Table.js

```diff
@@ -465,21 +479,35 @@
         fixedColTh
             .enter()
             .append("th")
-                .each(function (d) {
-                    var element = d3.select(this);
-                    element
-                        .append("span")
-                            .attr("class", "thText")
-                    ;
-                    element
-                        .append("span")
-                            .attr("class", "thIcon")
-                    ;
-                })
+            .each(function (d) {
+                var element = d3.select(this);
+                element
+                    .append("span")
+                        .attr("class", "thText")
+                ;
+                element
+                    .append("span")
+                        .attr("class", "thIcon")
+                ;
+            })
             .on("click", function (column, idx) {
-                context.headerClick(column, idx);
+                var tableDiv = context.tableDiv.node();
+                var clicked_elm = this;
+                var clicked_rect_before = clicked_elm.getBoundingClientRect();
+                var old_th_width = clicked_rect_before.width;
+                var old_th_left = clicked_rect_before.left;
+                context.headerClick(column, idx, function(){
+                    setTimeout(function(){
+                        var clicked_rect_after = clicked_elm.getBoundingClientRect();
+                        var new_th_width = clicked_rect_after.width;
+                        var new_th_left = clicked_rect_after.left;
+                        var l_diff = new_th_left - old_th_left;
+                        var w_diff = new_th_width - old_th_width;
+                        tableDiv.scrollLeft += l_diff + w_diff;
+                    },0);
+                });
             })
```

Comment:
Looks like duplicate code above - can you common it up (inside "headerClick" would be sensible)?